### PR TITLE
Add RoCE module

### PR DESCRIPTION
**Pull request by stalabi1**
##### SUMMARY

I added the RoCE module.

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

sonic_roce

##### OUTPUT

[regression-2024-05-15-15-22-53.html.pdf](https://github.com/ansible-collections/dellemc.enterprise_sonic/files/15327560/regression-2024-05-15-15-22-53.html.pdf)
[regression-2024-05-15-15-08-43.html.pdf](https://github.com/ansible-collections/dellemc.enterprise_sonic/files/15327561/regression-2024-05-15-15-08-43.html.pdf)
[diff_output.log](https://github.com/ansible-collections/dellemc.enterprise_sonic/files/15327567/diff_output.log)
[facts_gathering.log](https://github.com/ansible-collections/dellemc.enterprise_sonic/files/15327568/facts_gathering.log)
[show_running.txt](https://github.com/ansible-collections/dellemc.enterprise_sonic/files/15327570/show_running.txt)

##### Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility or have provided any relevant "breaking_changes" descriptions in a "fragment" file in the "changelogs/fragments" directory of this repository.

### CODE REVIEW

**Review by xhan-dell — COMMENTED** [review]
Looks pretty good, a few comments

File: plugins/modules/sonic_qos_buffer.py [hunk 1]
```diff
@@ -40,7 +40,8 @@
 - the configuration commands provided in this resource module. The buffer initialization will cause
 - the switch to reboot. See the "playbooks/common_examples/QoS_buffer_init.yaml" file
 - in this repo for an example playbook that initializes the buffers, waits for the reboot,
-- then proceeds with execution of QoS buffer configuration commands.
+- then proceeds with execution of QoS buffer configuration commands. Alternatively, make use of the
+- sonic_roce rseource module to enable RoCEv2 default buffer configuration.
```
Comment (kerry-meyer) [inline plugins/modules/sonic_qos_buffer.py:44]:
```suggestion
- sonic_roce resource module to enable RoCEv2 default buffer configuration.
```

**Review by kerry-meyer — CHANGES_REQUESTED** [review]
Thanks for all of your hard work to implement this difficult new configuration case.

The proposed code, new tests, and test results look good.

My only comments are for minor wording changes in documentation and comments.

**Review by kerry-meyer — APPROVED** [review]
Thanks for making these incremental revisions.

All issues are resolved now. The current proposed code, test coverage, and test results look good.

Approved.

**Pull request merged**